common/dal/src/interceptors/input_stream_interceptor.rs

```diff
@@ -40,9 +46,14 @@ impl futures::AsyncRead for InputStreamInterceptor {
         ctx: &mut std::task::Context<'_>,
         buf: &mut [u8],
     ) -> Poll<std::result::Result<usize, std::io::Error>> {
+        let start = Instant::now();
         let r = Pin::new(&mut self.inner).poll_read(ctx, buf);
+        let duration = start.elapsed();
+        self.cost_ms += duration.as_millis();
```

Comment:
Not sure this is the right place to calc the cost, need your check @dantengsky

Reply:
It would be better to take the `Poll::Pending`s into account as well.

i.e. if a sequence of responses returned by `poll_read` looks like:

<Pending, Pending,...Ready>

shall we take the time elapsed between the first Pending and the last Ready as `cost_ms`?

also, the `poll_seek` operations should be taken into account as well.

Reply:
Ok, I will address it again under your suggestion, thank you!